rmw_zenoh_cpp/src/detail/rmw_service_data.hpp

```diff
@@ -110,12 +111,15 @@ class ServiceData final
   const rmw_node_t * rmw_node_;
   // The Entity generated for the service.
   std::shared_ptr<liveliness::Entity> entity_;
-  // An owned keyexpression.
-  z_owned_keyexpr_t keyexpr_;
+
+  // A shared session
+  std::shared_ptr<zenoh::Session> sess_;
+  // The keyexpr string.
+  std::string keyexpr_;
   // An owned queryable.
-  z_owned_queryable_t qable_;
+  std::optional<zenoh::Queryable<void>> qable_;
```

Comment:
imo this should not be an optional. if we are unable to construct a `zenoh:::Queryable`, the `make()` function should fail.

Reply:
There is no public contructor for this class

Reply:
This is the same discussion as the earlier one.  Because of the design of these classes, I don't think we can do much about this.  But a comment would be helpful:

```suggestion
  // The Queryable *must* exist in order for anything in this ServiceData class, and hence rmw_zenoh_cpp, to work.
  // However, zenoh::Queryable does not have an empty constructor, so just declaring this as a zenoh::Queryable fails to compile.
  // We work around that by wrapping it in a std::optional, so the std::optional gets constructed at ServiceData constructor time,
  // and then we initialize qable_ later.  Note that the zenoh-cpp API declare_queryable() throws an
  // exception if it fails, so this should all be safe to do.
  std::optional<zenoh::Queryable<void>> qable_;
```